app/models/cart_item.rb

```diff
@@ -0,0 +1,6 @@
+class CartItem < ApplicationRecord
+  belongs_to :user 
+  belongs_to :product
+  # belongs_to :shop
+  has_one :shop, through: :product
```

Comment:
wrong association.
should be `belongs_to :shop`

Reply:
noted and fixed